src/Cake.Common.Tests/Unit/Tools/NuGet/List/NuGetListTests.cs

```diff
@@ -0,0 +1,250 @@
+
+            [Fact]
+            public void Should_Throw_If_Process_Has_A_Non_Zero_Exit_Code()
+            {
+                // Given
+                var fixture = new NuGetListFixture();
+                fixture.GivenProcessExitsWithCode(1);
+
+                // When
+                var result = Record.Exception(() => fixture.Run());
+
+                // Then
+                AssertEx.IsCakeException(result, "NuGet: Process returned an error (exit code 1).");
+            }
+
+            [Fact]
+            public void Should_Find_NuGet_Executable_If_Tool_Path_Not_Provided()
+            {
+                // Given
+                var fixture = new NuGetListFixture();
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("/Working/tools/NuGet.exe", result.Path.FullPath);
+            }
+
+            [Fact]
+            public void Should_Add_Mandatory_Arguments()
+            {
+                // Given
+                var fixture = new NuGetListFixture
+                {
+                    PackageId = "Cake"
+                };
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("list \"Cake\" -Verbosity Normal -NonInteractive", result.Args);
+            }
+
+            [Fact]
+            public void Should_Add_AllVersions_To_Arguments_If_True()
+            {
+                // Given
+                var fixture = new NuGetListFixture();
+                fixture.Settings.AllVersions = true;
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("list \"Cake\" -AllVersions -Verbosity Normal -NonInteractive", result.Args);
+            }
+
+            [Fact]
+            public void Should_Add_IncludeDelisted_To_Arguments_If_True()
+            {
+                // Given
+                var fixture = new NuGetListFixture();
+                fixture.Settings.IncludeDelisted = true;
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("list \"Cake\" -IncludeDelisted -Verbosity Normal -NonInteractive", result.Args);
+            }
+
+            [Fact]
+            public void Should_Add_Prerelease_To_Arguments_If_True()
+            {
+                // Given
+                var fixture = new NuGetListFixture();
+                fixture.Settings.Prerelease = true;
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("list \"Cake\" -Prerelease -Verbosity Normal -NonInteractive", result.Args);
+            }
+
+            [Fact]
+            public void Should_Add_Sources_To_Arguments_If_Set()
+            {
+                // Given
+                var fixture = new NuGetListFixture();
+                fixture.Settings.Source = new[] { "A;B;C" };
```

Comment:
Shouldn't this be `new[] { "A", "B", "C" };` in order to test that sources are concatenated corrected?

Reply:
@mholo65 If it is wrong here, then it's wrong in the same test in https://github.com/cake-build/cake/blob/develop/src/Cake.Common.Tests/Unit/Tools/NuGet/Install/NuGetInstallerTests.cs as well (there might be other places as well), since I shamelessly copied it from there. It would have to be tested in this test class as well since it's a different property of a different class. I depended on the Cake team for their expertise in this case.
That being said, I did some quick tests with the cli. The dot-comma separated syntax is apparently perfectly valid. So is adding multiple -Source arguments. So basically, this test only checks whether the value is appended to the -Source argument. The value just happens to be valid syntax for multiple repositories. We need additional tests to validate the concatenation, here and in the other nuget tests of the other commands. Good spotting though! 👍 
I'd suggest opening a separate issue to fix it in the multiple places at the same time. That should indeed be a quick improvement. The test in itself is not invalid, it just doesn't cover the concatenation part, but that goes for the other ones as well.  If you want, you can assign me immediately to that new issue to improve the tests regarding the concatenation. I'd be happy to add them. Suggestions?

Reply:
Note that multiple -Source arguments works, but it is not referenced/documented in https://docs.microsoft.com/en-us/nuget/tools/nuget-exe-cli-reference.